### PR TITLE
Custom events table

### PR DESCRIPTION
**Pull request by KennethLundberg**
Added the ability to create user defined custom events to add to a users schedule.

### CODE REVIEW

File: ucsdscheduleme/Controllers/HomeController.cs [hunk 1]
```diff
@@ -119,5 +118,86 @@ public IActionResult TypeAhead([FromBody] TypeAheadSearch search)
 
             return Json(suggestions);
         }
+
+        [HttpPost]
```
Comment (seannam) [inline ucsdscheduleme/Controllers/HomeController.cs]: function headers would be helpful in explaining what _data_ is

File: ucsdscheduleme/Repo/FormatRepo.cs [hunk 2]
```diff
@@ -108,7 +108,7 @@ private static BaseViewModel PopulateBaseForCourse(List<Section> sectionsForBase
         /// <param name="firstSection">The first section to pull the list of meetings from.</param>
         /// <param name="course">The course that all these sections are a part of.</param>
         /// <returns>A populated list of all calendar events.</returns>
-        private static List<CalendarEvent> PopulateBaseEvents(Section firstSection, Course course)
+        public static List<CalendarEvent> PopulateBaseEvents(Section firstSection, Course course)
```
Comment (seannam) [inline ucsdscheduleme/Repo/FormatRepo.cs]: why is this public now?

File: ucsdscheduleme/wwwroot/js/site.js [hunk 3]
```diff
@@ -1263,4 +1317,89 @@ function generateSchedule() {
             updateSchedule(response.courses);
         }
     }
+}
+
+function visibility_on(id) {
```
Comment (harishduwadi) [inline ucsdscheduleme/wwwroot/js/site.js]: Naming format.

File: ucsdscheduleme/wwwroot/js/site.js [hunk 4]
```diff
@@ -1263,4 +1317,89 @@ function generateSchedule() {
...
+    xhr.open("POST", url, true);
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    var send = {
+        "name": name,
+        "days": days,
+        "startTime": startTime,
+        "endTime": endTime
+    };
+
+    //TODO check valid input
```
Comment (git-ram) [inline ucsdscheduleme/wwwroot/js/site.js]: Did you check valid input?:/

File: ucsdscheduleme/Models/UserSection.cs [hunk 5]
```diff
@@ -8,7 +8,7 @@ namespace ucsdscheduleme.Models
     public class UserSection
     {
         public int Id { get; set; }
-        public int UserId { get; set; }
+        public string UserId { get; set; }
```
Comment (harishduwadi) [inline ucsdscheduleme/Models/UserSection.cs]: Confused on why userId was changed from int to string

File: ucsdscheduleme/wwwroot/js/site.js [hunk 6]
```diff
@@ -1263,4 +1317,89 @@ function generateSchedule() {
...
+                }
+            }
+
+            myApp.courses[text.courseId] = scheduleCourse;
+
+            updateSchedule(myApp.courses);
+        }
+    }
+}
+
```
Comment (git-ram) [inline ucsdscheduleme/wwwroot/js/site.js]: Maybe add function headers?

File: ucsdscheduleme/wwwroot/js/site.js [hunk 7]
```diff
@@ -1263,4 +1317,89 @@ function generateSchedule() {
...
+function visibility_on(id) {
+    var e = document.getElementById(id);
+    e.style.display = 'block';
+}
+
+function visibility_off(id) {
+    var e = document.getElementById(id);
+    e.style.display = 'none';
+}
+
```
Comment (git-ram) [inline ucsdscheduleme/wwwroot/js/site.js]: Function headers please!

File: ucsdscheduleme/wwwroot/js/site.js [hunk 8]
```diff
@@ -521,24 +499,80 @@ function addList(data)
...
+    console.log("data.id: " + event.id);
+    myApp.coursesToSchedule.push(event.id);
+}
+
+/**
+ * @description Adds the course selected from the search dropdown to the class itinerary.
+ * @param {HTMLElement} data class div selected to add
+ */
+function addCourse(data) {
+    console.log("HELLO");
```
Comment (harishduwadi) [inline ucsdscheduleme/wwwroot/js/site.js]: Might need to remove these two lines.

File: ucsdscheduleme/wwwroot/js/site.js [hunk 9]
```diff
@@ -521,24 +499,80 @@ function addList(data)
...
+    dropdown.style.display = "none";
+
+    // Create the element the add to the course list
+    var course = { "id": data.id, "courseAbbreviation": data.innerText };
+    addToScheduleList(course);
+
+    // Now clear the dropdown and repopulate it
+    typeAheadCallout(document.getElementById("search").value);
+}
+
```
Comment (git-ram) [inline ucsdscheduleme/wwwroot/js/site.js]: Function header missing

**Pull request merged**